### PR TITLE
EDE-469 Revert to old user registration flow and fix previous issues

### PR DESCRIPTION
**Pull request by mumarkhan999**
This PR is related to [EDE-469](https://edlyio.atlassian.net/browse/EDE-469)

**PR Description**
We implemented two different User Registration Flow for qverse.
1. In the first one, actual users were created from the uploaded CSV admission file and login credentials were sent to the user
2. In the second one, No actual users were created but the data of uploaded CSV admission file is saved and this data is used for the validation of the normal registration flow

Now as per client's recommendation we are reverting back to the first one.

### CODE REVIEW

File: openedx/features/qverse_features/registration/signals.py [hunk 1]
```diff
@@ -1,42 +1,46 @@
-"""
-Signals for qverse registration application.
-"""
```
Comment (danialmalik) [inline openedx/features/qverse_features/registration/signals.py]: I think this has been removed mistakenly.

File: openedx/features/qverse_features/registration/signals.py [hunk 2]
```diff
@@ -134,58 +196,139 @@ def _write_status_on_csv_file(filename, output_file_rows):
...
+            user = User.objects.get(email=self.email)
+            if user.username != self.regno:
+                self.errors.append('{} is already associated with another user account.'.format(self.email))
+        except User.DoesNotExist:
+            pass
+
+    def _validate_department_number(self):
+        try:
+            if not Department.objects.filter(number=self.department_id).exists():
+                self.errors.append('Department with this {} number does not exist.'.format(self.department_id))
```
Comment (danialmalik) [inline openedx/features/qverse_features/registration/signals.py]:
```suggestion
                self.errors.append('Department number {} does not exist.'.format(self.department_id))
```
???
Or use anything you prefer. I just find that "this <number> number" (say "this 10 number") odd.

File: openedx/features/qverse_features/registration/signals.py [hunk 3]
```diff
@@ -134,58 +196,139 @@ def _write_status_on_csv_file(filename, output_file_rows):
...
-        if int(row['levelid']) < 1 or int(row['levelid']) > MAX_LEVEL_CHOICES:
-            errors.append('Level ID must be greater than 0 and smaller than {}.'.format(MAX_LEVEL_CHOICES+1))
-    except ValueError:
-        errors.append('Level ID is not an integer value.')
+        Returns:
+            error (str): Contains error related to provided row
+        """
+        OPTIONAL_FIELDS = [
+            'mobile', 'othername', 'status', 'error', 'password'
+        ]
```
Comment (danialmalik) [inline openedx/features/qverse_features/registration/signals.py:256]: Wouldn't it be better to have a `REQUIRED_FIELDS` list instead?

File: openedx/features/qverse_features/registration/signals.py [hunk 4]
```diff
@@ -134,58 +196,139 @@ def _write_status_on_csv_file(filename, output_file_rows):
...
+    def _validate_department_number(self):
+        try:
+            if not Department.objects.filter(number=self.department_id).exists():
+                self.errors.append('Department with this {} number does not exist.'.format(self.department_id))
+        except ValueError:
+            self.errors.append('Please provide a valid integer value for department id.')
+
+    def _validate_registration_number(self):
+        if len(self.regno) > REGISTRATION_NUMBER_MAX_LENGTH:
+            self.errors.append('Registration number is more than {} characters long.'.format(REGISTRATION_NUMBER_MAX_LENGTH))
```
Comment (danialmalik) [inline openedx/features/qverse_features/registration/signals.py:305]: Should we also add a min length check here ?

File: openedx/features/qverse_features/registration/tasks.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    site = Site.objects.get(id=site_id)
+    context = {
+        'site_name': site.domain
+    }
+    for new_student in new_students:
+        user = None
+        try:
+            user = User.objects.get(username=new_student.get('regno'))
+        except User.DoesNotExist:
+            continue
```
Comment (danialmalik) [inline openedx/features/qverse_features/registration/tasks.py:41]: Log an error here ?

File: openedx/features/qverse_features/registration/signals.py (hunk 4 above)
Comment (mumarkhan999) [reply openedx/features/qverse_features/registration/signals.py:305]: Actually we are waiting for proper sample examples of the registration number. After that, all the validations related to the registration number will be applied.

**Review by danialmalik — APPROVED** [review]

**Review by imhassantariq — APPROVED** [review]

**Pull request merged**